### ISSUE TITLE
Transaction loading revamp

### ISSUE THREAD

**Issue by marekrjpolak** [trezor/trezor-suite#11306]
In order to be able to implement some transaction list improvements (e.g. filter out Ehtereum scam transactions or #7005), we have to decouple transaction pagination in Suite UI from api call pagination. The steps should be approximately following:

- remove pagination in `TransactionList` and make it infinitely scrolling list instead (with some clever UI maybe?)
  - fetch new transaction batch when needed
  - component virtualization/recyclation must be resolved first as the list can be quite long
  - page concept (`page` param in `fetchTransactionsThunks`) can be still preserved under the hood, but only directly following pages will be ever accessed
  - must work with tx filtering
- load all the account transactions eagerly instead of on demand
  - good for local-first, offline mode etc.
  - the same thing happens when filtering or exporting transactions
  - it can happen right after discovery (#7003), but it definitely shouldn't slow it down, so think about it as a second non-blocking stage of discovery?

With this, it should be possible to hide some transactions as they wouldn't need to correspond to selected page, and we won't have to deal with expected `undefined`'s in `transactionReducer` arrays because no page will ever be skipped.

**Comment by peter-sanderson** [comment]
Proof of concept done by Marek: https://github.com/trezor/trezor-suite/pull/11355

**Comment by vojtatranta** [comment]
Windsurf AI gave me this overlook of item rendering states. Not sure how accurate it is, but it is better than nothing.

### Transaction UI Elements and Conditions

| UI Element | TransactionItem | CoinjoinBatchItem | Conditions/Notes |
|------------|----------------|-------------------|------------------|
| Type Icon | ✓ | ✓ | TransactionItem: varies by tx type (send/receive/self/failed)<br>CoinjoinBatch: always "joint" type |
| Timestamp | ✓ | ✓ | Both show transaction time<br>TransactionItem: per transaction<br>CoinjoinBatch: shows latest tx time |
| Amount (Crypto) | ✓ | ✓ | TransactionItem: per transaction<br>CoinjoinBatch: sum of all transactions in batch |
| Amount (Fiat) | ✓ | ✓ | TransactionItem: current rate<br>CoinjoinBatch: historical rates for each tx |
| Fee | ✓ | - | Only for sent transactions<br>Multiple fees shown for multi-target txs |
| RBF Button | ✓ | - | Shown when:<br>- transaction.rbfParams exists<br>- network supports RBF<br>- no deadline exists |
| Warning Card | ✓ | - | Shows when isPending=true |
| Phishing Warning | ✓ | - | When transaction is flagged as phishing<br>Blurs all transaction details |
| Target Addresses | ✓ | - | Shows recipient addresses<br>Default limit of visible targets<br>Expandable for more targets |
| Collapsible View | - | ✓ | Shows individual coinjoin rounds<br>Each round shows its own amount |
| Transaction Count | - | ✓ | Total number of transactions in batch |
| Multiple Transactions | Shows as separate items | Groups related txs | TransactionItem: Multiple targets shown in expandable list<br>CoinjoinBatch: All related txs grouped in one collapsible item |
| Staking | ✓ | - | If the transaction was staked instantly / unstaked |
| Phishing | TODO | TODO | Levels of phishing - really scame, maybe scam)|
| Sent to my account | TODO | TODO | If the transaction was sent to one of "my" accounts|

### Special Layouts

**TransactionItem:**
- Single Row Layout (1 target):
  - Crypto amount in same row as heading
  - Fiat amount with address in second row
- Multi Row Layout (multiple targets):
  - Default shows first N targets
  - "Show more" button for additional targets
  - Each target shows its own amount

**CoinjoinBatchItem:**
- Always uses collapsible layout
- Batch summary shows total amounts
- Expandable to show individual rounds:
  - Each round has its own row
  - Shows round number and amount
  - Historical

**Comment by vojtatranta** [comment]
I checked the PR made by @marekrjpolak https://github.com/trezor/trezor-suite/pull/11355
TL;DR: I love it (great fetching API).

There are things that I don't understand / can't decide with my current knowledge (=no knowledge):
- what's the motivation / core of the unification of the staking and transaction unification - currently done with different lists
- the pagination is removed completely, making imcompatible (and unmergable basically) in the current state
- remove Transaction Actions component - and "replaced by" its content, I don't know the motivation here.

I know that @Nodonisko has been changing things around regarding fetching, making in nicer in Sept 2024. I am not sure why Marek's changes weren't used.

My proposal:
- Definitely use the useFetchTransactions hook, while giving it an ability to fetch by page (backwards-compatbile)
- keep the TransactionActions - don't know the reason of the removal
- keep the current "ununified" stake and transactions - for compatibility (up for discussion, I lack the context)

**Comment by vojtatranta** [comment]
Noticed: the scroll restore doesn't work properly, when clicking on the unloaded page, it gets scrolled way up.

**Comment by vojtatranta** [comment]
The progress of the issue and assignmed was somewhat tracked in the #16760 Thinking
- UI wise it was hard to implmenet "full scale infinite scroll" - so we left the old pagination and for cases when there are spam transaction to be hidden, we hid the "end page" coz it couldn't have been realiably calculated
- Then we agreeed that we need a switch there to enable this "feature" that one can keep the spam transaction hidden and then the whole feature is available
- currently, there is a tooltip and a switch

**Comment by bosomt** [comment]
**QA OK**

approved here https://github.com/trezor/trezor-suite/pull/16760#issuecomment-2740483736
